step-capstone/src/components/Sidebars/PreferencePopover.js

```diff
@@ -0,0 +1,66 @@
+import React, { useState, useEffect } from "react";
+import Popover from "@material-ui/core/Popover";
+import Typography from "@material-ui/core/Typography";
+
+export default function PreferencePopover(props) {
+  const [anchorEl, setAnchorEl] = React.useState(null);
+
+  useEffect(() => {
+    if (props.anchorEl !== anchorEl) {
+      setAnchorEl(props.anchorEl);
+    }
+  }, [props.anchorEl]);
+
+  useEffect(() => {
+    let coordinates = {lat: 0, lng: 0}
+    if (props.slots !== null) {
+      if (props.slots.prevSlot !== null && props.slots.nextSlot !== null) {
+        coordinates.lat = (props.slots.prevSlot.lat + props.slots.nextSlot.lat)/2;
+        coordinates.lgn = (props.slots.prevSlot.lgn + props.slots.nextSlot.lgn)/2
+      } else if (props.slots.prevSlot !== null) {
+        coordinates.lat = props.slots.prevSlot.lat;
+        coordinates.lgn = props.slots.prevSlot.lgn;
+      } else if (props.slots.nextSlot !== null ) {
+        coordinates.lat = props.slots.nextSlot.lat;
+        coordinates.lgn = props.slots.nextSlot.lgn;
+      } else {
+        //TODO: implement when merge code with Emmie's preference form and autocomplete
```

Comment:
My component actually is the suggestion bar popup at the bottom 😅

Reply:
I created a function in Trip that can be passed down to the timeline to pass up the selected trip information into the suggestions bar since it also needs userPreferences.

Reply:
Could you perhaps make the getting coordinates a function in the timeline component? that way when I pass the function down to you all you need to pass up is the coordinates, radius, and timerange

Reply:
Sure!